src/iceberg/sort_field.cc

```diff
@@ -0,0 +1,84 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+#include "iceberg/sort_field.h"
+
+#include <format>
+
+#include "iceberg/transform.h"
+#include "iceberg/type.h"
+#include "iceberg/util/formatter.h"
+
+namespace iceberg {
+
+namespace {
+/// \brief Get the relative sort direction name
+constexpr std::string_view ToString(SortDirection direction) {
+  switch (direction) {
+    case SortDirection::kAscending:
+      return "asc";
+    case SortDirection::kDescending:
+      return "desc";
+    default:
+      return "invalid";
+  }
+}
+
+/// \brief Get the relative null order name
+constexpr std::string_view ToString(NullOrder null_order) {
+  switch (null_order) {
+    case NullOrder::kFirst:
+      return "nulls-first";
+    case NullOrder::kLast:
+      return "nulls-last";
+    default:
+      return "invalid";
+  }
+}
+}  // namespace
+
+SortField::SortField(int32_t source_id, std::shared_ptr<TransformFunction> transform,
+                     SortDirection direction, NullOrder null_order)
+    : source_id_(source_id),
+      transform_(std::move(transform)),
+      direction_(direction),
+      null_order_(null_order) {}
+
+int32_t SortField::source_id() const { return source_id_; }
+
+std::shared_ptr<TransformFunction> const& SortField::transform() const {
+  return transform_;
+}
+
+SortDirection SortField::direction() const { return direction_; }
+
+NullOrder SortField::null_order() const { return null_order_; }
+
+std::string SortField::ToString() const {
+  return std::format(
+      "sort_field(source_id={}, transform={}, direction={}, null_order={})", source_id_,
```

Comment:
nit: it seems that the spec prints `transform` before `source_id`

Reply:
There a slight inconsistency in the spec, in `#sorting` section `source id` appears before `transform`, while in appendix `transform` before `source id`. iceberg-rust put `source_id` before `transform` in its display fmt[1], so I prefer to keep it as is ;)

[1] https://github.com/apache/iceberg-rust/blob/main/crates/iceberg/src/spec/sort.rs#L89-L97